tsdb/chunks/head_chunks_test.go

```diff
@@ -188,51 +185,41 @@ func TestHeadReadWriter_Truncate(t *testing.T) {
 		return mint
 	}
 
-	cutFile := func(i int) {
+	cutFile := func(i int) int64 {
 		testutil.Ok(t, hrw.CutNewFile())
-
-		mint := addChunk()
-
-		if i == startIndexAfter1stTruncation {
-			timeToTruncate = int64(mint)
-		} else if i == startIndexAfter2ndTruncation {
-			timeToTruncateAfterRestart = int64(mint)
-		}
-	}
-
-	// Cut segments.
-	for i := 1; i <= totalFiles; i++ {
-		cutFile(i)
+		return int64(addChunk())
 	}
 
 	// Verifying the files.
-	verifyFiles := func(remainingFiles, startIndex int) {
+	verifyFiles := func(remainingFiles []int) {
 		t.Helper()
 
 		files, err := ioutil.ReadDir(hrw.dir.Name())
 		testutil.Ok(t, err)
-		testutil.Equals(t, remainingFiles, len(files), "files on disk")
-		testutil.Equals(t, remainingFiles, len(hrw.mmappedChunkFiles), "hrw.mmappedChunkFiles")
-		testutil.Equals(t, remainingFiles, len(hrw.closers), "closers")
+		testutil.Equals(t, len(remainingFiles), len(files), "files on disk")
+		testutil.Equals(t, len(remainingFiles), len(hrw.mmappedChunkFiles), "hrw.mmappedChunkFiles")
+		testutil.Equals(t, len(remainingFiles), len(hrw.closers), "closers")
 
-		for i := 1; i <= totalFiles; i++ {
+		for _, i := range remainingFiles {
 			_, ok := hrw.mmappedChunkFiles[i]
-			if i < startIndex {
-				testutil.Equals(t, false, ok)
-			} else {
-				testutil.Equals(t, true, ok)
-			}
+			testutil.Equals(t, true, ok)
```

Comment:
Isn't `testutil.Assert` better for a bool?

Also, I'd add helpful information here, e.g.:
```Go
testutil.Assert(t, ok, "remaining file %d not in hrw.mmappedChunkFiles", i)
```

Reply:
Right!